packages/react-router-dom/index.tsx

```diff
@@ -851,18 +851,21 @@ function useSubmitImpl(fetcherKey?: string, routeId?: string): SubmitFunction {
   );
 }
 
-export function useFormAction(action = "."): string {
+export function useFormAction(action?: string): string {
   let routeContext = React.useContext(RouteContext);
   invariant(routeContext, "useFormAction must be used inside a RouteContext");
 
+  let location = useLocation();
   let [match] = routeContext.matches.slice(-1);
-  let { pathname, search } = useResolvedPath(action);
+  let path = useResolvedPath(action || location);
```

Comment:
I think this should still resolve the path based on `"."` if the action is undefined, no? Does that make a difference here? My reading of `resolveTo` makes me believe that it would if we were in a nested route situation.

Reply:
I think that puts us back in the same scenario though where we lose search params?  Down inside `useResolvedPath -> resolveTo -> parsePath` we end up with a `Path` of `{ pathname: '.' }` without any `search` or `hash`.  By passing the current `location` when no `action` is specified we preserve those throughout?  

Just to be safe, I added a few tests for the nested route scenario to confirm they're handled - and they break if we change to `let path = useResolvedPath(action || '.')`